pkg/dataobj/consumer/partition_processor.go

```diff
@@ -177,7 +193,19 @@ func (p *partitionProcessor) processRecord(record *kgo.Record) {
 			backoff.Wait()
 		}
 
-		if err := p.metastoreManager.UpdateMetastore(p.ctx, flushResult); err != nil {
+		backoff.Reset()
+		var objectPath string
+		for backoff.Ongoing() {
```

Comment:
I don't think it'd be unreasonable to move the backoff logic into the uploader to be consistent with the metastore, which also owns the logic for retrying. 

There's something to be said about whether the uploader/metastore manager have enough knowledge to be able to make the decision that they should be responsible for retrying, but it does make calling them simpler and consistency between the both would be nice, so having them both be responsible for retries seems fine to me.